packages/redis/src/lib/RedisProvider.ts

```diff
@@ -657,46 +697,58 @@ export class RedisProvider<StoredValue = unknown> extends JoshProvider<StoredVal
     return payload;
   }
 
+  protected async fetchVersion(): Promise<JoshProvider.Semver> {
+    const key = (await this.client.keys('*'))[0];
+    if (!key) return { major: 2, minor: 0, patch: 0 };
```

Comment:
```suggestion
    if (!key) return this.version;
```

Also, is there any way for us to get the version from the redis client rather then an entry from the client?

Reply:
Sadly not